beets/util/queue.py

```diff
@@ -138,6 +138,13 @@ def resume(self):
             else:
                 assert len(self._queue) == 0
 
+    def _is_full(self):
+        """Predicate factored out to increase readability of `put()`.
+
+        For internal use only, `self._mutex` must be held.
+        """
+        return self._maxsize == 0 or len(self._queue) >= self._maxsize
```

Comment:
If this (as the name implies) is meant to return True when the queue is full and can't currently accept any new elements, it seems intuitively that the `self._maxsize == 0 or` bit doesn't belong. That is, when `_maxsize == 0`, the queue is supposed to be infinite—i.e., *never* full, so this method should always return False. So I think it should instead be `self._maxsize > 0 and`?

Reply:
True, I got that wrong, `self._maxsize > 0 and [...]` is certainly correct.